File: Data Structures & Algorithms/design-hashmap/submission-0.py
class MyHashMap:

    def __init__(self):
        self.lim = 10_000
        self.st = [(-1, -1)] * self.lim

    def put(self, key: int, value: int) -> None:
        pos = key%self.lim
        
        for i in range(pos, self.lim):
            if self.st[i][0] == key: 
                self.st[i] = (key, value)
            elif self.st[i] == (-1, -1): break
        
        self.st[i] = (key,value)

    def get(self, key: int) -> int:
        pos = key%self.lim
        
        for i in range(pos, self.lim):
            if self.st[i][0] == key: return self.st[i][1]
            elif self.st[i] == (-1, -1): return -1

    def remove(self, key: int) -> None:
        pos = key%self.lim
        
        for i in range(pos, self.lim):
            if self.st[i][0] == key: 
                self.st[i] = (-1, -1)
            elif self.st[i] == (-1, -1): break
        


# Your MyHashMap object will be instantiated and called as such:
# obj = MyHashMap()
# obj.put(key,value)
# param_2 = obj.get(key)
# obj.remove(key)

File: Data Structures & Algorithms/implement-queue-using-stacks/submission-0.cpp
class MyQueue {
public:
    stack<int> q;

    MyQueue(): q() {}
    
    void push(int x) {
        q.push(x);
    }
    
    int pop() {
        vector<int> temp;
        while(q.size() != 1) {
            temp.push_back(q.top());
            q.pop();
        }

        int k = q.top();
        q.pop();

        for(int i = temp.size() - 1; i >= 0; i --) q.push(temp[i]);

        return k;
    }
    
    int peek() {
        vector<int> temp;
        while (q.size() > 1) {
            temp.push_back(q.top());
            q.pop();
        }
        int k = q.top();
        temp.push_back(k);
        q.pop();

        for(int i = temp.size() - 1; i >= 0; i --) q.push(temp[i]);
        return k;

    }
    
    bool empty() {
        return q.empty();
    }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */